### PR TITLE
OLED theme

### PR DESCRIPTION
**Pull request by EricBAndrews**
# Checklist
- [x] I have read CONTRIBUTING.md
- [x] I have described what this PR contains
- [x] If this PR alters the UI, I have attached pictures/videos
- [x] This PR addresses one or more open issues that were assigned to me:
      - Community feedback

# Pull Request Information

This PR adds an OLED theme that makes all backgrounds true black. To distinguish card entities, a light border is applied to the entity using the new `.paletteBorder` view modifier. This border has the same color as the palette-aware `Divider`.

To support this, `PaletteProviding` now defines a `bordered` boolean. `ColorPalette` defaults this to `false`, as it is a non-standard option. The divider color is now also a computed property of `PaletteProviding`

`List` and `Form` do not easily support a rounded border, so these views instead have a light shadow applied to them.

This PR also makes a couple minor changes:
- Padding on collapsed comments is adjusted to perfectly center the username within the collapsed comment when compact posts are off
- Padding on some account settings entities has been updated to play nice with the new shadow

### CODE REVIEW

File: Mlem/App/Utility/Extensions/Views/View Modifiers/View+PaletteBorder.swift [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+//
+//  Created by Eric Andrews on 2024-10-10.
+//
+
+import SwiftUICore
+
+import Foundation
+import SwiftUI
+
+private struct PaletteBorder: ViewModifier {
```
Comment (Sjmarf) [inline Mlem/App/Utility/Extensions/Views/View Modifiers/View+PaletteBorder.swift:13]: This needs to be applied to the crosspost list in `PostPage`

Edit: and the "New post" button in Profile, and the post in `ExpandedPostView`, and the Large mode website preview

File: Mlem/App/Views/Shared/Palette Components/Form.swift [hunk 2]
```diff
@@ -22,7 +22,9 @@ struct Form<Content: View>: View {
                 .tint(palette.accent)
                 .buttonStyle(PaletteButton())
         }
+        .listStyle(InsetListStyle())
         .scrollContentBackground(.hidden)
         .background(palette.groupedBackground)
+        .shadow(color: palette.primary.opacity(palette.bordered ? 0.2 : 0.0), radius: 5)
```
Comment (Sjmarf) [inline Mlem/App/Views/Shared/Palette Components/Form.swift:28]: This shadow shows up on `Form` sections for which we removed the background using `.listRowBackground`. Possibly fixable by applying another `.shadow` on those views or something?

File: Mlem/App/Configuration/Colors/OledPalette.swift [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        moderation: .cyan,
+        federatedFeed: .blue,
+        localFeed: .purple,
+        subscribedFeed: .red,
+        inbox: .purple,
+        accent: .blue,
+        neutralAccent: .gray,
+        colorfulAccents: [.orange, .pink, .blue, .green, .purple, .indigo, .mint],
+        commentIndentColors: [.red, .orange, .yellow, .green, .blue, .purple]
+    )
```
Comment (Sjmarf) [inline Mlem/App/Configuration/Colors/OledPalette.swift]: I wonder if it's better to do something like this? place

```swift
static let oled: ColorPalette = {
    var palette = ColorPalette.standard
    palette.background = .black
    // etc
    return palette
}()
```

That avoids defining the standard accents in more than one

**Review by Sjmarf — CHANGES_REQUESTED** [review]
Looks good! A few thoughts

File: Mlem/App/Globals/Definitions/Palette.swift [hunk 4]
```diff
@@ -67,6 +69,7 @@ enum PaletteOption: String, CaseIterable, Codable {
     var label: LocalizedStringResource {
         switch self {
         case .standard: "Default"
+        case .oled: "Oled"
```
Comment (Sjmarf) [inline Mlem/App/Globals/Definitions/Palette.swift]: "OLED" would be better than "Oled" imo

File: Mlem/App/Views/Shared/Palette Components/Form.swift (hunk 2 above)
Comment (EricBAndrews) [reply Mlem/App/Views/Shared/Palette Components/Form.swift:28]: Nice catch--this one was a little hacky, I had to move the content into the header of an empty Section. This does seem to fix the spacing issues that we were previously solving with negative padding.

File: Mlem/App/Utility/Extensions/Views/View Modifiers/View+PaletteBorder.swift (hunk 1 above)
Comment (EricBAndrews) [reply Mlem/App/Utility/Extensions/Views/View Modifiers/View+PaletteBorder.swift:13]: Nice catch!

File: Mlem/App/Configuration/Colors/OledPalette.swift (hunk 3 above)
Comment (EricBAndrews) [reply Mlem/App/Configuration/Colors/OledPalette.swift]: Yeah, that's cleaner 👍

File: Mlem/App/Views/Root/Tabs/Settings/AccountSettingsView.swift [hunk 5]
```diff
@@ -14,17 +14,17 @@ struct AccountSettingsView: View {
     
     var body: some View {
         Form {
-            Section {
+            // empty section disables background
+            Section {} header: {
```
Comment (Sjmarf) [inline Mlem/App/Views/Root/Tabs/Settings/AccountSettingsView.swift:18]: This makes the banner view thinner than it was before - possibly `.listRowInsets` would fix that?

We also use the empty section in the header of the "About" page

**Review by Sjmarf — CHANGES_REQUESTED** [review]

**Comment by Sjmarf** [comment]
The text fields in the sign-in flow also need `PaletteBorder`; I missed that one earlier :D

**Review by Sjmarf — APPROVED** [review]
LGTM! 💯

**Pull request merged**